tests/unit/test_configuration.py

```diff
@@ -662,3 +663,96 @@ def test_bad_duplicate_with_guardian_config(key: str) -> None:
                 },
             },
         )
+
+
+@pytest.mark.parametrize(
+    "noise_type, column, parameter, expected_value",
+    [
+        ("do_not_respond", None, None, 0.0145),
+        ("do_not_respond", None, "row_probability", 0.0145),
+        ("duplicate_with_guardian", None, "row_probability_in_households_under_18", 0.02),
+        ("leave_blank", "first_name", None, 0.01),
+        ("make_phonetic_errors", "first_name", "cell_probability", 0.01),
+        ("make_phonetic_errors", "first_name", "token_probability", 0.1),
+    ],
+)
+def test_working_general_noise_config_method(
+    noise_type: str, column: Optional[str], parameter: Optional[str], expected_value: float
+) -> None:
+    config = get_configuration()
+    noise_config = NoiseConfiguration(config)
+
+    value = noise_config.get_parameter_value(
+        "decennial_census", noise_type, column, parameter
+    )
+    assert value == expected_value
+
+
+@pytest.mark.parametrize(
+    "dataset, noise_type, column, parameter, error_msg",
+    [
+        ("fake_dataset", "noise_type", None, None, "fake_dataset was not found"),
+        (
+            "decennial_census",
+            "fake_noise_type",
+            None,
+            None,
+            "noise type fake_noise_type was not found",
+        ),
+        ("decennial_census", "duplicate_with_guardian", None, None, "multiple parameters"),
+        ("decennial_census", "do_not_respond", "fake_column", None, "cannot provide both"),
+        ("decennial_census", "leave_blank", None, None, "must provide a column name"),
+        ("decennial_census", "leave_blank", "fake_column", None, "fake_column was not found"),
+        (
+            "decennial_census",
+            "leave_blank",
+            "first_name",
+            "fake_parameter",
+            "fake_parameter was not found",
+        ),
+    ],
+)
+def test_breaking_general_noise_config_method(
+    dataset: str,
+    noise_type: str,
+    column: Optional[str],
+    parameter: Optional[str],
+    error_msg: str,
+) -> None:
+    config = get_configuration()
+    noise_config = NoiseConfiguration(config)
+
+    with pytest.raises(ValueError, match=error_msg):
+        value = noise_config.get_parameter_value(dataset, noise_type, column, parameter)
+
+
+@pytest.mark.parametrize("noise_type, expected_value", [("do_not_respond", 0.0145)])
```

Comment:
Is there a reason you're parameterizing this test even though you only have 1 case?

Reply:
I was anticipating that someone might request I add more use cases so I wanted to make it easy to setup haha. I think the structure amongst different noise types is similar enough that we wouldn't be testing anything meaningfully different if we added more cases but I'm open to hearing otherwise. If it's cleaner and we only want one case, I can remove the parametrization.